compiler/src/dotty/tools/dotc/transform/init/Objects.scala

```diff
@@ -183,7 +211,7 @@ object Objects:
     def show(using Context) = refs.map(_.show).mkString("[", ",", "]")
 
   /** A cold alias which should not be used during initialization. */
-  case object Cold extends Value:
+  case object Cold extends ValueElement:
```

Comment:
Consider making `Cold` a `Value`. Presumably any `RefSet` that contains `Cold` is equivalent to `Cold`. So we could make this invariant explicit: avoid ever putting a `Cold` in any `RefSet`, and replacing the `RefSet` with `Cold` as soon as we encounter a situation where we would want to put `Cold` into it.

Reply:
Oh, but then I see that there are some things (like `thisV`) that can be a `ValueElement` or `Cold`. So perhaps we want another type for that.